proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -344,7 +347,11 @@ private void handleConnectionException(RegisteredServer rs, @Nullable Component
     }
 
     Optional<RegisteredServer> getNextServerToTry() {
-        List<String> serversToTry = server.getConfiguration().getAttemptConnectionOrder();
+        if (serversToTry == null) {
+            Optional<List<String>> forcedHosts = getVirtualHost().map(vHost -> server.getConfiguration().getForcedHosts().get(vHost.getHostString()));
```

Comment:
This code will trigger the Optional Checker in Checker Framework. Prefer a `virtualHost != null` check and `server.getConfiguration().getForcedHosts().getOrDefault(virtualHost, server.getConfiguration().getAttemptConnectionOrder())`.